File: SDK-64bit/doc/_downloads/almotion_waituntilwalkisfinished.cpp
#include <iostream>
#include <alproxies/almotionproxy.h>
#include <qi/os.hpp>

int main(int argc, char **argv)
{
  if (argc < 2) {
    std::cerr << "Usage: motion_waituntilwalkisfinished pIp"
              << std::endl;
    return 1;
  }
  const std::string pIp = argv[1];

  AL::ALMotionProxy motion(pIp);

  std::vector<float> stiffnesses = motion.getStiffnesses("Body");
  if (stiffnesses[0] < 0.5f) {
    std::cerr << "Warning: this example will have no effect. "
              << "Robot must be stiff and standing." << std::endl;
    return 1;
  }

  // Example showing how to use waitUntilWalkIsFinished.
  // Start a walk
  float x = 0.10f;
  float y = 0.0f;
  float theta = 0.0f;
  motion.walkTo(x, y, theta);

  // Wait for it to finish
  motion.waitUntilWalkIsFinished();
  // Then do something else

  return 0;
}

File: SDK-64bit/doc/_downloads/almotion_waituntilwalkisfinished.py
import sys
from naoqi import ALProxy

if (len(sys.argv) < 2):
    print "Usage: 'python motion_waituntilwalkisfinished.py IP [PORT]'"
    sys.exit(1)

IP = sys.argv[1]
PORT = 9559
if (len(sys.argv) > 2):
    PORT = sys.argv[2]
try:
    proxy = ALProxy("ALMotion", IP, PORT)
except Exception,e:
    print "Could not create proxy to ALMotion"
    print "Error was: ",e
    sys.exit(1)

proxy.stiffnessInterpolation("Body", 1.0, 0.1)
proxy.walkInit()

# Example showing how to use waitUntilWalkIsFinished. 
 
# Start a walk 
x = 0.2 
y = 0.0 
theta = 0.0 
proxy.post.walkTo(x, y, theta) 
 
# Wait for it to finish  
proxy.waitUntilWalkIsFinished()
 
# Then do something else
print "Walk is finished"